econet_hvac_furnace.yaml

```diff
@@ -7,15 +7,64 @@ substitutions:
 packages:
   econet: !include econet_hvac.yaml
 
+econet:
+  on_datapoint_update:
+    - sensor_datapoint: HWSTATUS
+      datapoint_type: raw
+      then:
+        - lambda: |-
+            uint16_t airhandler_cfm_ = (x[13] << 8) + x[14];
+            uint16_t airhandler_rpm_ = (x[17] << 8) + x[18];
+            uint8_t heat_per_ = x[11];
+            uint8_t heat_stage_ = x[26];
+            uint16_t lh_lh_ = (x[129] << 8) + x[130];
+            uint16_t hh_lh_ = (x[132] << 8) + x[133];
+            uint8_t cool_stage_ = x[12];
+            float flame_sensor_ = ((float) x[33])/10;
+            float return_air_temperature_ = ((x[50] << 8) + x[51])/10.0;
+            float static_pressure_ = ((x[15] << 8) + x[16])/5280.0;
+            if(id(airhandler_cfm).state != airhandler_cfm_) {
+                id(airhandler_cfm).publish_state(airhandler_cfm_);
+            }
+            id(airhandler_rpm).publish_state(airhandler_rpm_);
+            id(lh_lh).publish_state(lh_lh_);
+            id(hh_lh).publish_state(hh_lh_);
+            id(heat_stage).publish_state(heat_stage_);
+            id(return_air_temperature).publish_state(return_air_temperature_);
+            id(flame_sensor).publish_state(flame_sensor_);
+            // Operating Mode
+            if(heat_per_ > 90)
+            {
+                id(operating_mode).publish_state("High Heat");
+            }
+            else if(heat_per_ > 0)
+            {
+                id(operating_mode).publish_state("Low Heat");
+            }
+            else if(cool_stage_ > 0)
+            {
+                id(operating_mode).publish_state("Cooling");
+            }
+            else if(airhandler_cfm_ > 0)
+            {
+                id(operating_mode).publish_state("Fan");
+            }
+            else
+            {
+                id(operating_mode).publish_state("Off");
+            }
+
```

Comment:
What's the reason we prefer to have this lambda here rather than putting something in the code? Even just a helper function that we could call from the lambda (if that's supported?)?

In addition to making the YAML bloated/complex, this code bypasses all of our static C++ analysis, IDE tooling for highlights, etc...

Reply:
I think with the simplifications identified by @tronikos, this is greatly simplified.